test/test_pub_sub.py.in

```diff
@@ -260,22 +259,24 @@ def generate_test_description(ready_fn):
         node_relay_test,
     ]
 
-    return LaunchDescription(nodes + [
-        OpaqueFunction(function=lambda context: ready_fn()),
-    ]), locals()
+    return (
+        LaunchDescription(nodes + [launch_testing.actions.ReadyToTest()]),
+        locals())
 
 
 class PerformanceTestResults(unittest.TestCase):
 
     def test_results_@TEST_NAME_PUB_SUB@(
-            self, proc_info, nodes,
+            self, nodes,
             performance_log_cpumem_pub, performance_log_cpumem_sub,
             performance_log_prefix_pub, performance_log_prefix_sub):
         self.addCleanup(_cleanUpLogs, performance_log_prefix_pub + '*')
         self.addCleanup(_cleanUpLogs, performance_log_prefix_sub + '*')
         self.addCleanup(_cleanUpLogs, performance_log_cpumem_pub)
         self.addCleanup(_cleanUpLogs, performance_log_cpumem_sub)
 
+        proc_info = self.proc_info
```

Comment:
This is a nit, but since `proc_info` is only used once below, I think it would be better to change line 281 to just use `self.proc_info`.

Reply:
Fixed in 9f03288